### PR TITLE
feat: resolve bare import by default

### LINKED ISSUE #282: How export rolldown resolver?
### Description 

The rollup external node packages by default, it need `@rollup/plugin-resolve` to do it. The rolldown maybe have some ways to do it, it need discuss at now.

1. enable node packages by default,  add `resolve` option to control the resolver behavior.
2. follow rollup and add `resolve` option to enable native plugin `plugin-resolve`.

Hope get some advices, thanks.

### PR DESCRIPTION
**Pull request by underfin**
### Description

close https://github.com/rolldown-rs/rolldown/issues/282

### Test Plan

---

### CODE REVIEW

**Comment by underfin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #280** <a href="https://app.graphite.dev/github/pr/rolldown-rs/rolldown/280?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #283** <a href="https://app.graphite.dev/github/pr/rolldown-rs/rolldown/283?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #284** <a href="https://app.graphite.dev/github/pr/rolldown-rs/rolldown/284?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/rolldown-rs/rolldown/284?utm_source=stack-comment).

File: crates/rolldown/src/bundler/stages/scan_stage.rs [hunk 1]
```diff
@@ -86,15 +86,10 @@ impl<Fs: FileSystem + Default + 'static> ScanStage<Fs> {
         )
         .await
         {
-          Ok(r) => {
-            let Some(info) = r else {
-              return Err(BuildError::unresolved_entry(specifier));
```
Comment (underfin) [inline crates/rolldown/src/bundler/stages/scan_stage.rs:91]: BuildError::unresolved_entry already be handled.

**Review by hyf0 — APPROVED** [review]

**Comment by hyf0** [comment]
It's would be good to add a test to ensure this behavior.

```js
import {} from 'is-plain-object'
```

should works.

You could refer to https://github.com/rolldown-rs/rolldown/tree/main/crates/rolldown/tests/fixtures/resolve/node_modules_as_entries.

---edited

Just found out the test case is confusing. You could refer to this fixed one: https://github.com/rolldown-rs/rolldown/pull/285

**Comment by underfin** [comment]
I add an `external` option to fix some tests which use this behavior, I think it already could be tested.

**Pull request merged**